macros/grants/grants.yml

```diff
@@ -0,0 +1,75 @@
+version: 2
+
+macros:
+  - name: grant_database_ownership_access
+    description: This macro grants ownership privilege to the specific roles within the database
+    docs:
+      show: true
+    arguments:
+      - name: rolename
+        type: string
+        description: Name of the role eg- ANALYST
+
+  - name: grant_database_usage_access
+    description: This macro grants database usages privilege to the specific roles within the database. To get access on the database, the role needs to have usage privilege in the database.
+    docs:
+      show: true
+    arguments:
+      - name: rolename
+        type: string
+        description: Name of the role eg- ANALYST
+
+  - name: grant_schema_ownership_access
+    description: This macro grants ownership privilege inside the schema to a specific role. 
+    docs:
+      show: true
+    arguments:
+      - name: schemas
+        type: string
+        description: Name of the schema
+      - name: rolename
+        type: string
+        description: Name of the role eg- ANALYST
+      - name: include_future_grants
+        type: bool
+        description: Grants future privilege to the objects FUTURE VIEWS, FUTURE MATERIALIZED VIEWS, FUTURE TABLES, and FUTURE EXTERNAL TABLES. Default is false
+
+  - name: grant_schema_read_access
+    description: This macro grants read-only access to the specified role inside the schema. 
```

Comment:
This macro grants read-only access to the specified rolename inside the schema.

Reply:
done